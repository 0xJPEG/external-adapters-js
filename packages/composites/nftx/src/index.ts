import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import { makeConfig } from './config'
import * as types from './endpoint'

const NAME = 'NFTX_VAULT_PRICE'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)
export { NAME, makeExecute, makeConfig, server, types }
