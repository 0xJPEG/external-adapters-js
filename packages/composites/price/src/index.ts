import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'ANCHOR_PROTOCOL'

export = { NAME, makeConfig, makeExecute, ...expose(NAME, makeExecute()) }
