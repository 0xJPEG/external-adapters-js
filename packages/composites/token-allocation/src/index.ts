import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './endpoint'
import { makeEndpointSelector } from './adapter'
import { makeConfig } from './config'
import * as types from './types'

const NAME = 'Token-Allocation'
const { server } = expose(NAME, makeExecute())

export { NAME, types, makeExecute, makeConfig, server, makeEndpointSelector }
