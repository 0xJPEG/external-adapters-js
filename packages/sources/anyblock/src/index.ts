import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig } from './config'

const NAME = 'ANYBLOCK'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
