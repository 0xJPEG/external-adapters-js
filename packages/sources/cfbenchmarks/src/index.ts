import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute, makeWSHandler } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'

const { server } = expose(NAME, makeExecute(), makeWSHandler(), endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
