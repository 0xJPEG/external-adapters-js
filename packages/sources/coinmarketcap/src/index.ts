import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, endpointSelector } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'

const server = expose(NAME, makeExecute(), undefined, endpointSelector).server

export { NAME, endpoints, makeConfig, makeExecute, server }
