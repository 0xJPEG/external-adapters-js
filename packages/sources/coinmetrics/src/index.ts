import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute, makeWSHandler } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'
import * as types from './endpoint'

const { server } = expose(NAME, makeExecute(), makeWSHandler(), endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server, types }
