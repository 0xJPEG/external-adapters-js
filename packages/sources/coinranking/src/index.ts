import { expose } from '@chainlink/ea-bootstrap'
import { makeConfig, NAME } from './config'
import * as endpoints from './endpoint'
import { endpointSelector, makeExecute } from './adapter'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
