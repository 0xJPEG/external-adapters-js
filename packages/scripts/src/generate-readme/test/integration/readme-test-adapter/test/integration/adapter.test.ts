import { AdapterRequest } from '@chainlink/types'
import { util } from '@chainlink/ea-bootstrap'
import http from 'http'
import nock from 'nock'
import request from 'supertest'
import { server as startServer } from '../../src/index'
import { mockBalanceResponse, mockMarketcapResponse, mockPriceResponse } from './fixtures'

let oldEnv: NodeJS.ProcessEnv

describe('readme test adapter', () => {
  let server: http.Server
  const req = request('localhost:8080')

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    process.env.CACHE_ENABLED = 'false'
    process.env.API_KEY = process.env.API_KEY || 'mock-api-key'
    if (util.parseBool(process.env.RECORD)) {
      nock.recorder.rec()
    }
    server = await startServer()
  })
  afterAll((done) => {
    process.env = oldEnv
    if (util.parseBool(process.env.RECORD)) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
    server.close(done)
  })

  describe('balance endpoint', () => {
    it('should reply with success', async () => {
      const balanceRequest: AdapterRequest = {
        id: '1',
        data: {
          endpoint: 'balance',
          addresses: [{ address: '3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1' }],
          dataPath: 'addresses',
        },
      }

      mockBalanceResponse()
      const response = await req
        .post('/')
        .send(balanceRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('marketcap endpoint', () => {
    it('should reply with success', async () => {
      const marketcapRequest: AdapterRequest = {
        id: '1',
        data: {
          coin: 'BTC',
          market: 'USD',
          endpoint: 'marketcap',
          coinid: 'Bitcoin',
        },
      }

      mockMarketcapResponse()
      const response = await req
        .post('/')
        .send(marketcapRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })

    it('should reply with success when called with `mc`', async () => {
      const mcRequest: AdapterRequest = {
        id: '1',
        data: {
          coin: 'BTC',
          market: 'USD',
          endpoint: 'mc',
          coinid: 'Bitcoin',
        },
      }

      mockMarketcapResponse()
      const response = await req
        .post('/')
        .send(mcRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('price endpoint', () => {
    it('should reply with success', async () => {
      const priceRequest: AdapterRequest = {
        id: '1',
        data: {
          base: 'BTC',
          quote: 'USD',
          endpoint: 'price',
        },
      }

      mockPriceResponse()
      const response = await req
        .post('/')
        .send(priceRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })

    it('should reply with success when called with `convert`', async () => {
      const priceRequest: AdapterRequest = {
        id: '1',
        data: {
          base: 'BTC',
          quote: 'USD',
          endpoint: 'convert',
        },
      }

      mockPriceResponse()
      const response = await req
        .post('/')
        .send(priceRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
