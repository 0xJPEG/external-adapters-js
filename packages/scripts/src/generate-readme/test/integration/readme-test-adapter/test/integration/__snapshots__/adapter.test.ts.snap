// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readme test adapter balance endpoint should reply with success 1`] = `
Object {
  "data": Object {
    "responses": Array [
      Object {
        "description": "Successful request",
        "payload": Object {
          "address": Object {
            "address": "3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1",
          },
          "blockNumber": "693286",
          "blockchainId": "408fa195a34b533de9ad9889f076045e",
          "timestamp": "2021-07-29T20:54:39.000Z",
          "timestampNanoseconds": 0,
          "value": "2188",
        },
        "status": 200,
        "title": "OK",
      },
    ],
    "result": Array [
      Object {
        "address": "3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1",
        "balance": "2188",
        "chain": "mainnet",
        "coin": "btc",
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "address": "3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1",
      "balance": "2188",
      "chain": "mainnet",
      "coin": "btc",
    },
  ],
  "statusCode": 200,
}
`;

exports[`readme test adapter marketcap endpoint should reply with success 1`] = `
Object {
  "data": Object {
    "marketcap": 1000000000,
    "result": 1000000000,
  },
  "jobRunID": "1",
  "result": 1000000000,
  "statusCode": 200,
}
`;

exports[`readme test adapter marketcap endpoint should reply with success when called with \`mc\` 1`] = `
Object {
  "data": Object {
    "marketcap": 1000000000,
    "result": 1000000000,
  },
  "jobRunID": "1",
  "result": 1000000000,
  "statusCode": 200,
}
`;

exports[`readme test adapter price endpoint should reply with success 1`] = `
Object {
  "data": Object {
    "price": 123456,
    "result": 123456,
  },
  "jobRunID": "1",
  "result": 123456,
  "statusCode": 200,
}
`;

exports[`readme test adapter price endpoint should reply with success when called with \`convert\` 1`] = `
Object {
  "data": Object {
    "price": 123456,
    "result": 123456,
  },
  "jobRunID": "1",
  "result": 123456,
  "statusCode": 200,
}
`;
